source/particle/generator/probability_density_function.cc

```diff
@@ -186,20 +184,25 @@ namespace aspect
                   }
                 try
                   {
-                    const Point<dim> p_unit = this->get_mapping().transform_real_to_unit_cell(it, particle_position);
-                    if (GeometryInfo<dim>::is_inside_unit_cell(p_unit)) break;
+                    const Point<dim> p_unit = this->get_mapping().transform_real_to_unit_cell(cell, particle_position);
+                    if (GeometryInfo<dim>::is_inside_unit_cell(p_unit))
+                      break;
                   }
                 catch (typename Mapping<dim>::ExcTransformationFailed &)
                   {
                     // The point is not in this cell. Do nothing, just try again.
                   }
-                num_tries++;
+                iteration++;
               }
-            AssertThrow (num_tries < 100, ExcMessage ("Couldn't generate particle (unusual cell shape?)."));
+            AssertThrow (iteration < 100,
```

Comment:
This is brittle because you use the same magic number (100) in two different places, the `while` statement and the assertion. Create a variable `const unsigned int max_iterations` or similar that you have to define only once so that the right thing will continue to happen if someone decides to ever change the 100.

I think you have the same code in multiple places -- change it everywhere.